### DISCUSSION TITLE
Walmart and Target Issues

### DISCUSSION THREAD

**Discussion by xmpierce1** [Prince25/StockAlertBot#64] [Q&A] [answered]
Hey Prince,

I seem to be getting an error with Walmart and the log says:

Error: Request failed with status code 412
    at createError (C:\StockAlertBot-3.3\node_modules\axios\lib\core\createError.js:16:15)
    at settle (C:\StockAlertBot-3.3\node_modules\axios\lib\core\settle.js:17:12)
    at IncomingMessage.handleStreamEnd (C:\StockAlertBot-3.3\node_modules\axios\lib\adapters\http.js:260:11)
    at IncomingMessage.emit (events.js:327:22)
    at IncomingMessage.EventEmitter.emit (domain.js:467:12)
    at endReadableNT (internal/streams/readable.js:1327:12)
    at processTicksAndRejections (internal/process/task_queues.js:80:21) {
  config: {
    url: 'https://www.walmart.com/ip/Xbox-Series-X/443574645',
    method: 'get',
    headers: {
      Accept: 'application/json, text/plain, */*',
      'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.14; rv:10.0) Gecko/20100101 Firefox/62.0'

This is with this URL : https://www.walmart.com/ip/Xbox-Series-X/443574645

_____

Target is still throwing an error:

TypeError: Cannot read property 'textContent' of undefined
    at Timeout.target [as _onTimeout] (file:///C:/StockAlertBot-3.3/stores/target.js:69:150)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)

on URL : https://www.target.com/p/xbox-series-x-console/-/A-80790841
and
https://www.target.com/p/playstation-5-console/-/A-81114595

Can you see if you can try to replicate ?

**Comment by Prince25** [comment]
Hi again. Is the Walmart error occurring all the time? Are you using proxies? 
I'll run it on my end tomorrow.

**Comment by xmpierce1** [comment]
No proxies.. it seems to happen from the start of the check.

**Comment by Prince25** [comment, marked as answer]
I ran Walmart for some times and it's working correctly for me: 
![image](https://user-images.githubusercontent.com/28243638/113217939-85295d00-9233-11eb-9e4b-cb0c510a19f4.png)
[Status code 412](http://man.hubwiz.com/docset/HTTP.docset/Contents/Resources/Documents/developer.mozilla.org/en-US/docs/Web/HTTP/Status/412.html) "indicates that access to the target resource has been denied." There could be many reasons for this. Are you able to access Walmart in a browser just fine? You can try replacing line 56 in `Walmart.js` with the following and see if that helps:
```js
else options = { 
	headers: {
		'accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9',
		'accept-language': 'en-US,en;q=0.9',
		'cache-control': 'no-cache',
		'pragma': 'no-cache',
		'sec-ch-ua-mobile': '?0',
		'sec-fetch-dest': 'document',
		'sec-fetch-mode': 'navigate',
		'sec-fetch-site': 'none',
		'sec-fetch-user': ?1,
		'service-worker-navigation-preload': 'true',
		'upgrade-insecure-requests': '1',
		'User-Agent': USER_AGENTS[Math.floor(Math.random() * USER_AGENTS.length)] 
	}
}
```

As for Target, it also works just fine for me: 
![image](https://user-images.githubusercontent.com/28243638/113218797-02a19d00-9235-11eb-8ecb-6a85fc214bd4.png)
But I've pushed an update (786f99c0701cc0c944ee47c735bf5e258c3673a7) which should keep it from giving a random error. See if it shows product info error and how often.

**Comment by xmpierce1** [comment]
I will try it.. It really seems cosmetic for Target become PS5 dropped this morning and the alart worked perfect.. Walmart is an odd one though.
